File: src/main/java/ru/melonhell/bulletphysics/bullet/collision/space/cache/block/impl/DirectBlockCache.java
package ru.melonhell.bulletphysics.bullet.collision.space.cache.block.impl;

import lombok.Getter;
import ru.melonhell.bulletphysics.bullet.collision.space.MinecraftSpace;
import ru.melonhell.bulletphysics.bullet.collision.space.cache.block.BlockCache;
import ru.melonhell.bulletphysics.bullet.collision.space.cache.block.data.BlockDataUtils;
import ru.melonhell.bulletphysics.bullet.collision.space.cache.block.data.PhysicsBlockData;
import ru.melonhell.bulletphysics.nms.wrappers.BlockPos;

public class DirectBlockCache implements BlockCache {
    @Getter
    private final MinecraftSpace space;
    private final BlockDataUtils blockDataUtils;

    public DirectBlockCache(MinecraftSpace space, BlockDataUtils blockDataUtils) {
        this.space = space;
        this.blockDataUtils = blockDataUtils;
    }

    @Override
    public PhysicsBlockData getBlockData(BlockPos blockPos) {
        return blockDataUtils.getPhysicsBlockData(blockPos.getBlock(space.getWorld()));
    }
}

File: src/main/java/ru/melonhell/bulletphysics/bullet/collision/space/cache/block/impl/DirectBlockCache.kt
package ru.melonhell.bulletphysics.bullet.collision.space.cache.block.impl

import ru.melonhell.bulletphysics.bullet.collision.space.MinecraftSpace
import ru.melonhell.bulletphysics.bullet.collision.space.cache.block.BlockCache
import ru.melonhell.bulletphysics.bullet.collision.space.cache.block.data.BlockDataUtils
import ru.melonhell.bulletphysics.bullet.collision.space.cache.block.data.PhysicsBlockData
import ru.melonhell.bulletphysics.nms.wrappers.BlockPos

class DirectBlockCache(
    override val space: MinecraftSpace,
    private val blockDataUtils: BlockDataUtils
) : BlockCache {
    override fun getBlockData(blockPos: BlockPos): PhysicsBlockData {
        return blockDataUtils.getPhysicsBlockData(blockPos.getBlock(space.world))
    }
}